### ISSUE TITLE
cross_val_score() shows nan had tried error_score='raise', but no error

### ISSUE THREAD

**Issue by elcolie** [skorch-dev/skorch#854]
Here is my model.

```python
import torch
import torch.nn as nn
import torch.nn.functional as F
import sklearn


class LitDoubleFCNetwork(torch.nn.Module):
    def __init__(self, input_size: int = 600, hidden_size=600, num_layers: int = 2, dropout: float = 0.2,
                 output_size: int = 5):
        super().__init__()
        self.input_size = input_size
        self.hidden_size = hidden_size
        self.num_layers = num_layers
        self.dropout = dropout
        self.output_size = output_size

        self.rnn = nn.GRU(input_size=self.input_size, hidden_size=self.hidden_size, num_layers=self.num_layers,
                          batch_first=True,
                          dropout=self.dropout)
        # self.linear = nn.Linear(in_features=self.hidden_size, out_features=self.hidden_size)
        self.fc = nn.Linear(in_features=self.hidden_size, out_features=self.output_size)

    # def forward(self, x: torch.Tensor) -> torch.Tensor:
    def forward(self, *args, **kwargs) -> torch.Tensor:
        """Forward function."""
        
        _keys = [_ for _ in kwargs.keys()]
        first_key = _keys[0]
        batch_size = len(kwargs[first_key])
        n_array = []

        for j in range(batch_size):
            row_array = []
            for i in range(self.input_size): 
                row_array.append(
                    kwargs.get(str(i))[j]
                )
            n_array.append(row_array)
        x = torch.FloatTensor(n_array)
        
        x, hn = self.rnn(x)
        # x = self.linear(x)
        # x = F.relu(x)
        x = self.fc(x)
        x = F.relu(x)
        output = F.log_softmax(x, dim=1)
        return output
```

Then I wrap it by `NeuralNetClassifier`
```python
from skorch import NeuralNetClassifier
net = NeuralNetClassifier(
    LitDoubleFCNetwork,
    max_epochs=5,
    lr=0.001,
    iterator_train__shuffle=True,
)
```
Read the data source
```python
import pandas as pd

X_res = pd.read_csv("products/CO2_simplified_X_res.csv")
y_res = pd.read_csv("products/CO2_simplified_y_res.csv")
```
Run `cross_val_score`
```python
from sklearn.model_selection import cross_val_score
cross_val_score(net, X_res, y_res.values.ravel(), error_score='raise')
```

**Expected result**<br>
Valid numerical value

**As is**<br>
`nan` is shown up
```bash
 epoch    train_loss    valid_acc    valid_loss       dur
-------  ------------  -----------  ------------  --------
      1           nan       0.0912           nan  456.0337
```

**Comment by elcolie** [comment]
After try add/remove linear layer in my network. It just gone!

**Comment by BenjaminBossan** [comment]
For other people coming across this:

1. The `nan` occurred because `log_softmax` was used, when by default `NeuralNetClassifier` expects `softmax`. Best practice is to use _no output nonlinearity_  in conjunction with `criterion=CrossEntropyLoss`.
2. `cross_val_score` did not raise an error because it does not monitor (or even know about) the `train_loss` or `valid_loss`. In theory, at the _end of the training step_ (not _during the training_), it would calculate the error and then raise if it is `nan`. In your case, you did not specify the `scoring` argument to `cross_val_score`, so it would try to call `net.score(X, y)`, which would result in an error. You should pass a valid `scoring` argument.
